File: Models/Psychiatrist.cs
namespace Project.Models;


public class Psychiatrist(string name, string specialization) : Doctor(name, specialization)
{
    private List<KeyValuePair<int, int>> _schedule = new();

    public void AddSchedule(KeyValuePair<int, int> time)
    {
        _schedule.Add(time);
    }

    public override void DisplayInfo()
    {
        Console.WriteLine("Name: " + GetName());
        Console.WriteLine("Specialization: " + Specialization);
        Console.WriteLine("Schedule:");
        foreach (var scheduleItem in _schedule)
        {
            Console.WriteLine($"{GetDayOfWeek(scheduleItem.Key)} at {scheduleItem.Value}:00");
        }
    }

    public override void BookAppointment(Data time)
    {
        Console.WriteLine($"Appointment booked with {Name} at {time}");
    }

    public override List<KeyValuePair<int, int>> GetSchedule()
    {
        return _schedule;
    }
}

File: lab1/psychiatrist.cpp
#include "psychiatrist.h"
#include <iostream>
#include <string>
#include <vector>
#include "datetime.h"   

using namespace std;

Psychiatrist::Psychiatrist(const string& name, const string& specialization) : Doctor(name, specialization) {}

void Psychiatrist::addSchedule(const pair<int, int> time) {
    schedule.push_back(time);
}

void Psychiatrist::displayInfo() const {
    cout << "Name: " << getName() << endl;
    cout << "Specialization: " << specialization << endl;
    cout << "Schedule:" << endl;
    for (const auto& scheduleItem : schedule) {
        cout << getDayOfWeek(scheduleItem.first) << " at " << scheduleItem.second << ":00" << endl;
    }
}

void Psychiatrist::bookAppointment(const DateTime& time) {
    cout << "Appointment booked with " << getName() << " at " << formatDate(time) << endl;
}

const vector<pair<int, int>>& Psychiatrist::getSchedule() const {
    return schedule;
}